### DISCUSSION TITLE
V1 free vs V3 paid, really?

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [acheong08/ChatGPT#1072] [General]
I noticed the following description in [README](https://github.com/acheong08/ChatGPT#v1-standard-chatgpt):

![image](https://user-images.githubusercontent.com/11155854/223426823-707afde8-df80-41f6-b489-e5c8872b681a.png)

I think all of them are paid. As the proverb goes: "There is no such thing as a pie falling from the sky".

Regards,
Zhao

**Comment by acheong08** [comment]
`https://chat.openai.com` is free but protected by Cloudflare. Once Cloudflare bypassed, it is free. However, if OpenAI is made aware of the bypass, it will no longer be free. To keep it available, I run a proxy server so that the bypass is not revealed. 

Since I'm paying for the server costs which allow it to continue being free, yes, it is technically paid, just by me rather than you.

**Reply by acheong08** [reply]
But I'm running out of money to pay for the server

**Reply by hongyi-zhao** [reply]
> `https://chat.openai.com` is free but protected by Cloudflare. Once Cloudflare bypassed, it is free. However, if OpenAI is made aware of the bypass, it will no longer be free. To keep it available, I run a proxy server so that the bypass is not revealed.
> 
> Since I'm paying for the server costs which allow it to continue being free, yes, it is technically paid, just by me rather than you.

I think that you're using a vps running on the cloud to do job, but I'm not aware what's the price plan in your scenario.

**Reply by acheong08** [reply]
> I think that you're using a vps running on the cloud to do job

yes

> but I'm not aware what's the price plan in your scenario.

I get bandwidth of around 700GB / day at the cost of $0.01 per GB.

4 servers $20 and a load balancer $12 (per month)

**Reply by hongyi-zhao** [reply]
$20 + $12 (per month) = $32 (per month) = 222,28 Chinese Yuan (per month), which is more expensive than my proxy.

![image](https://user-images.githubusercontent.com/11155854/223462320-67340f2f-150e-4d05-a2e1-2d10105aa0be.png)

**Reply by acheong08** [reply]
It comes from having to load balance that much traffic. 1 server would crash within minutes

**Reply by acheong08** [reply]
And the most expensive is the bandwidth. It costs $7 a day which is $210 a month

**Reply by acheong08** [reply]
Which is about 1,457.90 CNY / month

**Reply by acheong08** [reply]
This will literally bankrupt me soon

**Reply by hongyi-zhao** [reply]
Really, for individuals, this is indeed a very expensive expenditure.

**Reply by acheong08** [reply]
It's only because I provide a public service. If it was for personal use, $5 per month would suffice. the issue is that I have over 119k unique users of the unofficial API

**Reply by hongyi-zhao** [reply]
If so, why do you still say that `V3 paid`? IMO, it's also protected by your server.

**Reply by acheong08** [reply]
It's not within my server. V3 is direct access to https://api.openai.com/v1/chat/completions

**Reply by hongyi-zhao** [reply]
TBF, `V3` is faster and cheaper than `V1`. So, it seems that it's time for you to stop providing `V1` services personally.

**Reply by acheong08** [reply]
Agreed. I need to somehow convince everyone that has V1 as a dependency though

**Reply by hongyi-zhao** [reply]
I admire your spirit of service for free.

**Reply by chelishchev** [reply]
> It's only because I provide a public service. If it was for personal use, $5 per month would suffice. the issue is that I have over 119k unique users of the unofficial API

Is it possible to share code of proxy server? To deploy it byself?

**Reply by acheong08** [reply]
> Is it possible to share code of proxy server? To deploy it byself?

The issue with that is the fact that the method would be blocked by OpenAI if they are made aware. 

https://github.com/fuergaosi233/chatgpt-proxy-server is an open source implementation but also with a few details missing required to run
